pkg/cmd/repo/clone/clone.go

```diff
@@ -165,6 +166,12 @@ func cloneRun(opts *CloneOptions) error {
 	// Load the repo from the API to get the username/repo name in its
 	// canonical capitalization
 	canonicalRepo, err := api.GitHubRepo(apiClient, repo)
+	if err != nil && repositoryIsFullName {
+		return err
+	}
+	// Try to find the repo in the user's organizations, since the user did not
+	// specify the owner.
+	canonicalRepo, err = repoFromUserOrgs(apiClient, repo)
```

Comment:
What do you think about falling back to a prompt like the one used in `gh repo create` that has the user selecting the owner rather than going with the first organization that happens to have a repo by the same name?

```shell
$ gh repo create                   
? What would you like to do? Create a new repository on GitHub from scratch
? Repository name asdfasdf
? Repository owner  [Use arrows to move, type to filter]
  0ctocat
  Avocado-Extra-Charge
  OpenSourceFellows
  alltheavo
> andyfeller
  andyfeller-demos
  andyfeller-gei-migrate-org-test
  casa-vega
  cli
  client-apps-ec-testing
  docs
  github
  ...
```

Reply:
I think this is a great idea! I also considered this as an option as well, I think this is a good idea since if someone is part of many repositories we won't have to make `N` queries, but it still solves the pain point. I can implement this!

Reply:
Would you be open to using a `huh` form for this? Or should we stick to `go-survey`? We can style the `huh` form to pretty much look like `go-survey` so everything looks consistent.

Reply:
<img width="912" alt="image" src="https://github.com/cli/cli/assets/42545625/a06b5ef3-cc17-4d94-bff6-d0b5eda9f1a3">

Reply:
I'm very, very interested in the UX that styling `huh` like `go-survey` results in. One of the challenges with moving over to `huh` is that moving every command over, and changing the UX is very risky technically and for our users. If we could match `huh` to `go-survey` on a single command, that could be very informative and perhaps pave a path for us to move other prompts one by one.